packages/live-status-gateway/src/topics/segmentsTopic.ts

```diff
@@ -0,0 +1,94 @@
+import { Logger } from 'winston'
+import { WebSocket } from 'ws'
+import { DBRundownPlaylist } from '@sofie-automation/corelib/dist/dataModel/RundownPlaylist'
+import { WebSocketTopicBase, WebSocketTopic, CollectionObserver } from '../wsHandler'
+import { DBSegment } from '@sofie-automation/corelib/dist/dataModel/Segment'
+import { PlaylistHandler } from '../collections/playlist'
+import { groupByToMap } from '@sofie-automation/corelib/dist/lib'
+import { unprotectString } from '@sofie-automation/shared-lib/dist/lib/protectedString'
+import { SegmentsHandler } from '../collections/segmentsHandler'
+import isShallowEqual from '@sofie-automation/shared-lib/dist/lib/isShallowEqual'
+
+interface SegmentStatus {
+	id: string
+	rundownId: string
+	name: string
+}
+
+export interface SegmentsStatus {
+	event: 'segments'
+	rundownPlaylistId: string | null
+	segments: SegmentStatus[]
+}
+
+export class SegmentsTopic
+	extends WebSocketTopicBase
+	implements WebSocketTopic, CollectionObserver<DBSegment[]>, CollectionObserver<DBRundownPlaylist>
+{
+	public observerName = SegmentsTopic.name
+	private _activePlaylist: DBRundownPlaylist | undefined
+	private _segments: DBSegment[] = []
+	private _orderedSegments: DBSegment[] = []
+
+	constructor(logger: Logger) {
+		super(SegmentsTopic.name, logger)
+	}
+
+	addSubscriber(ws: WebSocket): void {
+		super.addSubscriber(ws)
+		this.sendStatus(new Set<WebSocket>().add(ws))
+	}
+
+	sendStatus(subscribers: Set<WebSocket>): void {
```

Comment:
```suggestion
		this.sendStatus([ws])
	}

	sendStatus(subscribers: Set<WebSocket> | WebSocket[]): void {
```
Suggestion: this is a bit simpler and cleaner